src/main/ruby/truffleruby/core/io.rb

```diff
@@ -2158,6 +2158,23 @@ def set_encoding(external, internal=nil, options=undefined)
     self
   end
 
+  def set_encoding_by_bom
+    unless binmode?
+      raise ArgumentError, 'ASCII incompatible encoding needs binmode'
+    end
+
+    if !Primitive.nil?(internal_encoding)
```

Comment:
Just `if internal_encoding` would be fine here, since `internal_encoding` can only be `nil` or an `Encoding`, same for the `external_encoding` check below

Reply:
Makes sense. Done!